apps/bajour/components/website-builder-overwrites/blocks/teaser-slider/teaser-slider.tsx

```diff
@@ -0,0 +1,194 @@
+import 'keen-slider/keen-slider.min.css'
+
+import {css, styled, useMediaQuery, useTheme} from '@mui/material'
+import {
+  ApiV1,
+  BuilderTeaserGridBlockProps,
+  hasBlockStyle,
+  isFilledTeaser,
+  isTeaserGridBlock
+} from '@wepublish/website'
+import {useKeenSlider} from 'keen-slider/react'
+import {allPass} from 'ramda'
+import {useState} from 'react'
+
+import {TeaserSlide} from './teaser-slide'
+
+export const isTeaserSlider = (block: ApiV1.Block): block is ApiV1.TeaserGridBlock =>
+  allPass([hasBlockStyle('Slider'), isTeaserGridBlock])(block)
+
+export const SliderContainer = styled('div')`
+  display: grid;
+  position: relative;
+  padding-top: ${({theme}) => theme.spacing(2)};
+  padding-bottom: ${({theme}) => theme.spacing(2)};
+  gap: ${({theme}) => theme.spacing(5)};
+
+  ${({theme}) => theme.breakpoints.up('sm')} {
+    padding-top: ${({theme}) => theme.spacing(3)};
+    padding-bottom: ${({theme}) => theme.spacing(3)};
+  }
+
+  ${({theme}) => theme.breakpoints.up('md')} {
+    padding-top: ${({theme}) => theme.spacing(5)};
+  }
+
+  ${({theme}) => theme.breakpoints.up('xl')} {
+    padding-top: ${({theme}) => theme.spacing(7)};
+  }
+`
+
+const SliderInnerContainer = styled('div')``
+
+const SliderInnerBackground = styled('div')`
+  position: absolute;
+  z-index: -1;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  background: #feede8;
+
+  ${({theme}) => theme.breakpoints.up('sm')} {
+    left: calc(100% / 12);
+    right: calc(100% / 12);
+  }
+
+  ${({theme}) => theme.breakpoints.up('md')} {
+    left: calc((100% / 12) * 2);
+    right: calc((100% / 12) * 2);
+  }
+`
+
+const BallContainer = styled('div')`
+  display: flex;
+  justify-content: center;
+  gap: ${({theme}) => theme.spacing(1)};
+`
+
+const Ball = styled('button', {
+  shouldForwardProp: prop => prop !== 'isActive'
+})<{isActive: boolean}>`
+  appearance: none;
+  border: none;
+  width: ${({theme}) => theme.spacing(2)};
+  height: ${({theme}) => theme.spacing(2)};
+  background: ${({theme}) => theme.palette.common.white};
+  border-radius: 50%;
+  cursor: pointer;
+
+  :focus {
+    outline: none;
+  }
+
+  ${({theme, isActive}) =>
+    isActive &&
+    css`
+      background: ${theme.palette.error.main};
+    `}
+`
+
+const useSlidesPerView = () => {
+  const theme = useTheme()
+
+  const sm = useMediaQuery(theme.breakpoints.up('sm'), {
+    ssrMatchMedia: () => ({matches: false})
+  })
+
+  const md = useMediaQuery(theme.breakpoints.up('md'), {
+    ssrMatchMedia: () => ({matches: false})
+  })
+
+  const lg = useMediaQuery(theme.breakpoints.up('lg'), {
+    ssrMatchMedia: () => ({matches: false})
+  })
+
+  const xl = useMediaQuery(theme.breakpoints.up('xl'), {
+    ssrMatchMedia: () => ({matches: false})
+  })
+
+  if (xl) {
+    return 2.8
```

Comment:
decimals allows to show more of the previews, they shouldn't fully show (else it would be 3) but to be partially cut off.